File: SigreAppMobile/Hermes/common-ui/src/main/java/pe/com/hermes/common/ui/ConfirmDialog.java
package pe.com.hermes.common.ui;

import android.content.Context;
import com.google.android.material.dialog.MaterialAlertDialogBuilder;

/**
 * Dialogo Si/No generico — equivalente moderno de ConfirmDialog.java de FastSales.
 */
public final class ConfirmDialog {

    public interface OnAccion {
        void run();
    }

    private ConfirmDialog() {
    }

    public static void mostrar(Context context, String titulo, String mensaje, String textoSi, String textoNo,
                                boolean cancelable, OnAccion onSi, OnAccion onNo) {
        MaterialAlertDialogBuilder builder = new MaterialAlertDialogBuilder(context)
                .setMessage(mensaje)
                .setCancelable(cancelable)
                .setPositiveButton(textoSi, (dialog, which) -> {
                    dialog.dismiss();
                    if (onSi != null) onSi.run();
                })
                .setNegativeButton(textoNo, (dialog, which) -> {
                    dialog.dismiss();
                    if (onNo != null) onNo.run();
                });
        if (titulo != null && !titulo.trim().isEmpty()) builder.setTitle(titulo);
        builder.show();
    }

    public static void mostrar(Context context, String titulo, String mensaje, boolean cancelable, OnAccion onSi) {
        mostrar(context, titulo, mensaje, "Sí", "Cancelar", cancelable, onSi, null);
    }
}

File: SigreAppMobile/Hermes/common-ui/src/main/java/pe/com/hermes/common/ui/ConfirmDialog.kt
package pe.com.hermes.common.ui

import android.content.Context
import com.google.android.material.dialog.MaterialAlertDialogBuilder

/**
 * Diálogo Sí/No genérico — equivalente moderno de ConfirmDialog.java de FastSales
 * (que usaba una interfaz `Action` con métodos vacíos por defecto), reescrito con
 * lambdas de Kotlin y MaterialAlertDialogBuilder en vez de AlertDialog.Builder puro.
 */
object ConfirmDialog {

    fun mostrar(
        context: Context,
        mensaje: String,
        titulo: String? = null,
        textoSi: String = "Sí",
        textoNo: String = "Cancelar",
        cancelable: Boolean = true,
        onSi: () -> Unit,
        onNo: (() -> Unit)? = null,
    ) {
        val builder = MaterialAlertDialogBuilder(context)
            .setMessage(mensaje)
            .setCancelable(cancelable)
            .setPositiveButton(textoSi) { dialog, _ -> dialog.dismiss(); onSi() }
            .setNegativeButton(textoNo) { dialog, _ -> dialog.dismiss(); onNo?.invoke() }

        if (!titulo.isNullOrBlank()) builder.setTitle(titulo)
        builder.show()
    }
}
